Scripts/Tracking/Follow/ObjectFollow.cs

```diff
@@ -0,0 +1,217 @@
+﻿namespace VRTK.Core.Tracking.Follow
+{
+    using UnityEngine;
+    using UnityEngine.Events;
+    using System;
+    using VRTK.Core.Data.Attribute;
+    using VRTK.Core.Data.Enum;
+    using VRTK.Core.Process;
+    using VRTK.Core.Tracking.Follow.Modifier;
+
+    /// <summary>
+    /// The ObjectFollow mirrors the transform properties of another object based on the given FollowModifier.
+    /// </summary>
+    public class ObjectFollow : SourceTargetProcessor
+    {
+        [Header("Object Follow Settings")]
+
+        [UnityFlag]
+        [Tooltip("The transform properties to follow.")]
+        public TransformProperties follow;
+        [Tooltip("The follow modifier mechanic to apply.")]
+        public FollowModifier followModifier = null;
+
+        /// <summary>
+        /// The ObjectFollowUnityEvent emits an event with a source Transform and a target Transform.
+        /// </summary>
+        [Serializable]
+        public class ObjectFollowUnityEvent : UnityEvent<Transform, Transform> { };
+
+        [Header("Object Follow Events")]
+
+        /// <summary>
+        /// The BeforeProcessed event is emitted before any processing.
+        /// </summary>
+        public ObjectFollowUnityEvent BeforeProcessed;
+        /// <summary>
+        /// The AfterProcessed event is emitted after all processing is complete.
+        /// </summary>
+        public ObjectFollowUnityEvent AfterProcessed;
+        /// <summary>
+        /// The BeforeTransformUpdated event is emitted before the Transform is updated.
+        /// </summary>
+        public ObjectFollowUnityEvent BeforeTransformUpdated;

```

Comment:
Naming: It's `Processed` everywhere, but these things are `Updated` instead. Shall we rename these and the methods like `UpdatePosition` etc. to use `Process(ed|ing)` instead?

Reply:
hmm, you're not technically processing the position it is actually updating the position.

Process doesn't necessarily mean something will be updated.

And in the point of these events, a `Process` of `Updates` happens and before the `Process` starts you get an event and at the end of the `Process` you get an event, then in between you have `Updates`